wolfcrypt/src/rsa.c

```diff
@@ -3632,12 +3630,22 @@ int wc_RsaPSS_CheckPadding_ex(const byte* in, word32 inSz, byte* sig,
     }
 
 #ifdef WOLFSSL_PSS_LONG_SALT
-    if (sigCheck != NULL) {
-        XFREE(sigCheck, NULL, DYNAMIC_TYPE_RSA_BUFFER);
+    if (sigCheck != sigCheckBuf) {
```

Comment:
sigCheck can still be NULL I believe.

Reply:
It can be NULL, but XFREE already handles NULL checking. However I went ahead and added the check anyways.